wasm_utils/src/api_serialization/commit.rs

```diff
@@ -0,0 +1,29 @@
+use holochain_core_types::{cas::content::Address, hash::HashString};
+
+/// Struct for input data received when Commit API function is invoked
+#[derive(Deserialize, Default, Debug, Serialize)]
+pub struct CommitEntryArgs {
+    pub entry_type_name: String,
+    pub entry_value: String,
+}
+#[derive(Deserialize, Serialize, Default, Debug)]
+pub struct CommitEntryResult {
+    pub address: Address,
+    pub validation_failure: String,
+}
+
+impl CommitEntryResult {
+    pub fn success(address: Address) -> CommitEntryResult {
+        CommitEntryResult {
+            address,
+            validation_failure: String::from(""),
+        }
+    }
+
+    pub fn failure(validation_failure: String) -> CommitEntryResult {
+        CommitEntryResult {
+            address: HashString::from(""),
+            validation_failure,
+        }
+    }
+}
```

Comment:
I believe there is a better way to do this:
 - On success: Return the CommitEntryOutput struct. 
 - On failure: Return a ErrorOutput struct.

 On wasm side, if deserializing of CommitEntryOutput fails, then deserialize an ErrorOutput.

Reply:
I prefer that too

Reply:
I thought about that but decided not to since I would either have to roll my own `try_deserialize!()` in `hdk::commit_entry()` or put commit/validation concern into the general `try_deserialize!()`. A fallback deserialization retry feels adequate for error handling, but a validation failure is not an error and specific to commit.

I struggled a bit with the implicit complexity we have here already. Given that we expect other HDK implementations to be added in the future, I'd rather keep this bridge simple and explicit rather then complex and implicit.